src/rs.cpp

```diff
@@ -174,6 +174,7 @@ struct rs2_error
 
 rs2_error *rs2_create_error(const char* what, const char* name, const char* args, rs2_exception_type type) BEGIN_API_CALL
 {
+    LOG_ERROR( "[" << name << "][" << rs2_exception_type_to_string( type ) << "] " << what << ": " << args );
```

Comment:
Why do we need to log it?
Errors through `translate_exception` would probably already be logged elsewhere.
Don't know if we should log new, user created exceptions.

Reply:
We do not.
We relied on recoverable_exception to log, but that was exactly what I was trying to remove: it's too low-level and too numerous when an exception is entirely reasonable.
Here, an error is generated from our high-level API. So we log it.

I agree it's questionable... But if we do not want to "dirty" the logs, that's what we need to do.
You prefer leaving the recoverable_exception log there? With the changes in the emitter/hdr handling, maybe it'll clean up the log...

Reply:
Even with recoverable_exception, it did not cover `runtime_error`, which also gets thrown.
I.e., I don't think the right way to do it is via recoverable_exception.

Reply:
OK